packages/lesswrong/server/vulcan-lib/apollo-server/authentication.tsx

```diff
@@ -57,7 +57,7 @@ const passwordAuthStrategy = new GraphQLLocalStrategy(async function getUserPass
     return done(null, false, { message: 'Incorrect password.' });
   }
   
-  const match = await comparePasswords(password, user.services.password.bcrypt);
+  const match = await comparePasswords(password, user.services.password?.bcrypt);
```

Comment:
@jimrandomh

Reply:
The first check for services.password.bcrypt prevents the potential security vulnerability, but note that you'll need to use null-safe access (`?.`) to avoid the same bug as we had previously.